homeassistant/components/recorder/core.py

```diff
@@ -314,73 +331,52 @@ def _async_event_filter(self, event: Event) -> bool:
         # Unknown what it is.
         return True
 
-    def do_adhoc_purge(self, **kwargs: Any) -> None:
-        """Trigger an adhoc purge retaining keep_days worth of data."""
-        keep_days = kwargs.get(ATTR_KEEP_DAYS, self.keep_days)
-        repack = cast(bool, kwargs[ATTR_REPACK])
-        apply_filter = cast(bool, kwargs[ATTR_APPLY_FILTER])
-
-        purge_before = dt_util.utcnow() - timedelta(days=keep_days)
-        self.queue.put(PurgeTask(purge_before, repack, apply_filter))
-
-    def do_adhoc_purge_entities(
-        self, entity_ids: set[str], domains: list[str], entity_globs: list[str]
-    ) -> None:
-        """Trigger an adhoc purge of requested entities."""
-        entity_filter = generate_filter(domains, list(entity_ids), [], [], entity_globs)
-        self.queue.put(PurgeEntitiesTask(entity_filter))
-
     def do_adhoc_statistics(self, **kwargs: Any) -> None:
         """Trigger an adhoc statistics run."""
         if not (start := kwargs.get("start")):
             start = statistics.get_start_time()
-        self.queue.put(StatisticsTask(start))
+        self.queue_task(StatisticsTask(start))
 
```

Comment:
We currently only call this in tests.  We should probably move this into the the test code instead